src/pages/orders/components/OrdersDateSelection/OrdersDateSelection.tsx

```diff
@@ -0,0 +1,76 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+import { DatePicker } from '@/components';
+import { LabelPairedCalendarRangeMdRegularIcon } from '@deriv/quill-icons';
+import { useTranslations } from '@deriv-com/translations';
+import { Input, useDevice } from '@deriv-com/ui';
+import { OrdersDateSelectionFullPage } from '../OrdersDateSelectionFullPage';
+import './OrdersDateSelection.scss';
+
+type TOrdersDateSelectionProps = {
+    fromDate: string | null;
+    setFromDate: Dispatch<SetStateAction<string | null>>;
+    setToDate: Dispatch<SetStateAction<string | null>>;
+    toDate: string | null;
+};
+
+const OrdersDateSelection = ({ fromDate, setFromDate, setToDate, toDate }: TOrdersDateSelectionProps) => {
+    const { isMobile } = useDevice();
+    const { localize } = useTranslations();
+    const [isDateSelectionOpen, setIsDateSelectionOpen] = useState(false);
+
+    const getMaxDate = () => {
+        if (toDate) {
+            return new Date(toDate);
+        }
+        return new Date();
+    };
```

Comment:
nit: just my personal preference since this is reduced to one line but both are equally fine

```suggestion
    const getMaxDate = () => toDate ? new Date(toDate) : new Date();
```